gcloud/monitoring/client.py

```diff
@@ -0,0 +1,189 @@
+                    default inferred from the environment.
+
+    :type credentials: :class:`oauth2client.client.OAuth2Credentials` or
+                       :class:`NoneType`
+    :param credentials: The OAuth2 Credentials to use for the connection
+                        owned by this client. If not passed (and if no ``http``
+                        object is passed), falls back to the default inferred
+                        from the environment.
+
+    :type http: :class:`httplib2.Http` or class that defines ``request()``
+    :param http: An optional HTTP object to make requests. If not passed, an
+                 ``http`` object is created that is bound to the
+                 ``credentials`` for the current object.
+    """
+
+    _connection_class = Connection
+
+    def query(self,
+              metric_type=Query.DEFAULT_METRIC_TYPE,
+              end_time=None,
+              days=0, hours=0, minutes=0):
+        """Construct a query object for listing time series.
+
+        Example::
+
+            >>> query = client.query(minutes=5)
+            >>> print query.as_dataframe()  # Requires pandas.
+
+        :type metric_type: string
+        :param metric_type: The metric type name. The default value is
+            ``"compute.googleapis.com/instance/cpu/utilization"``, but
+            please note that this default value is provided only for
+            demonstration purposes and is subject to change. See the
+            `supported metrics`_.
+
+        :type end_time: :class:`datetime.datetime` or None
+        :param end_time: The end time (inclusive) of the time interval
+            for which results should be returned, as a datetime object.
+            The default is the start of the current minute.
+
+            The start time (exclusive) is determined by combining the
+            values of  ``days``, ``hours``, and ``minutes``, and
+            subtracting the resulting duration from the end time.
+
+            It is also allowed to omit the end time and duration here,
+            in which case
+            :meth:`~gcloud.monitoring.query.Query.select_interval`
+            must be called before the query is executed.
+
+        :type days: integer
+        :param days: The number of days in the time interval.
+
+        :type hours: integer
+        :param hours: The number of hours in the time interval.
+
+        :type minutes: integer
+        :param minutes: The number of minutes in the time interval.
+
+        :rtype: :class:`~gcloud.monitoring.query.Query`
+        :returns: The query object.
+
+        :raises: :exc:`ValueError` if ``end_time`` is specified but
+            ``days``, ``hours``, and ``minutes`` are all zero.
+            If you really want to specify a point in time, use
+            :meth:`~gcloud.monitoring.query.Query.select_interval`.
+
+        .. _supported metrics: https://cloud.google.com/monitoring/api/metrics
+        """
+        return Query(self, metric_type,
```

Comment:
In general I avoid documenting `raises` if the exception isn't directly raised by the method in question. So here, I'd really on the raises section for `Query.__init__` rather than having one of my own.

This is a defense against docs growing apart over time.